app/Http/Controllers/Assets/AssetsController.php

```diff
@@ -314,9 +308,17 @@ public function update(ImageUploadRequest $request, $assetId = null)
         $asset->status_id = $request->input('status_id', null);
         $asset->warranty_months = $request->input('warranty_months', null);
         $asset->purchase_cost = $request->input('purchase_cost', null);
-        $asset->asset_eol_date  = request('asset_eol_date', null);
+        if ($request->filled('purchase_date') && !$request->filled('asset_eol_date')) {
+            $asset->asset_eol_date = Carbon::parse($request->input('purchase_date'))->addMonths($asset->model->eol)->format('Y-m-d');
+        } elseif ($request->filled('asset_eol_date')) {
+           $asset->explicit_eol = request('explicit_eol', null); 
```

Comment:
So if we're going to do that, why wouldn't we just go ahead and make `explicit_eol` a boolean instead of a secondary date field? If it's going to end up containing the same value I mean.

Reply:
Then we have one column to check for EOL date, and `explicit_eol` is just a quick boolean check to determine whether or not we update it when the model's eol or purchase date are modified. (Could also use it on the GUI to display a little icon noting that it's an explicitly set date)